theme/js/helpers.js

```diff
@@ -0,0 +1,71 @@
+import RemoveMarkdown from "remove-markdown";
+import markdown from "./markdown";
+import dayjs from "dayjs/esm";
+import { useI18n } from 'vue-i18n';
+
+export const truncate = (val, length = 300) => {
+  if (typeof val !== "string") return;
+  return val.length > length ? val.slice(0, length) + "…" : val; //TODO, maybe® : properly truncate words
+};
+
+export const excerpt = (val, length = 300) => {
+  if (typeof val !== "string") return;
+  return truncate(RemoveMarkdown(val), length);
+};
+
+export const filesize = (val) => {
+    const { t } = useI18n();
+    const suffix = t('B')
+    const units = ['', 'K', 'M', 'G', 'T', 'P', 'E', 'Z']
+    for (let unit of units) {
+        if (Math.abs(val) < 1024.0) {
+          return `${val.toFixed(1)}${unit}${suffix}`
+        }
+        val /= 1024.0
+    }
+    return `${val.toFixed(1)}Y${suffix}`
+}
+
+/**
+ * Format date based on locale.
+ * @param {Date | string} date
+ * @returns {string}
+ */
+export const formatDate = (date, format = 'LL') => {
+  return dayjs(date).format(format);
+}
+
+/**
+ * Format date as relative from now.
+ * It displays "today" or dayjs.fromNow content, based on date.
+ * @param {Date | string} date
+ * @returns {string}
+ */
+export const formatFromNow = (date) => {
+  const { t } = useI18n();
+  const today = dayjs().startOf("date");
+  const dateWithoutTime = dayjs(date).startOf("date");
+  if(today.isSame(dateWithoutTime)) {
```

Comment:
Yes, we can even choose when it rounds to the next unit. We'll encounter some challenges because udata dates are sent as UTC by the API even if they are not.